[Yingming] iP

Yingming23 opened the pull request:
# Duke
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke allows you to save tasks to be done in a list, so that you never have to struggle to remember them again. It's 

- text-based
- easy-to-learn
- ~~fast~~ super fast to use

All you need to do is:

1. Download it from [here](https://github.com/Yingming23/ip/releases/tag/A-Jar).
2. Double click it.
3. Add your tasks.
4. Let it manage your tasks for you. :octocat: 

And it is ***FREE***!

Features:

- [x] Managing tasks
- [x] Finding tasks on a certain date
- [x] Finding tasks by a certain word

If you are a Java programmer, you can use it to practise Java too. Here is the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Yingming23 committed:
Duke.java Add Greet, Echo, Exit

Yingming23 committed:
Duke.java Add Add, List

Yingming23 committed:
Duke.java Add Mark as Done

Yingming23 committed:
Duke.java Add ToDos, Events, Deadlines

Yingming23 committed:
Add input text and expected text

Yingming23 committed:
Added exceptions and separate files for classes

Yingming23 committed:
Added delete function

Yingming23 committed:
Reformat all classes to be more in line with OOP principles

Yingming23 committed:
Added save function

Yingming23 committed:
Merge commit '150b75b45da82828ae48d28bbc648096a52cb468'

Yingming23 committed:
Merge branch 'branch-Level-7'

Yingming23 committed:
Added date and times, as well as DateCommand to find tasks on a specific date

Yingming23 committed:
Merge branch 'branch-Level-8'

Yingming23 committed:
Added Ui class and renamed some classes to better reflect requirements

Yingming23 committed:
Divided classes into packages

Yingming23 committed:
Imported gradle

Yingming23 committed:
Added checkstyle

Yingming23 committed:
Added JUnit tests

Yingming23 committed:
Added JAR

Yingming23 committed:
Added JavaDocs

Yingming23 committed:
Changed names to comply with coding standards

Yingming23 committed:
Added Find Command

Yingming23 committed:
Merge branch 'branch-A-JavaDoc'

Yingming23 committed:
Merge branch 'branch-A-CodingStandard'

Yingming23 committed:
Resolved merge conflict

Yingming23 committed:
Changed wildcard imports

senwei01 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,146 @@
+package duke;
```
I really like how you have multiple outputs for different cases such as show that a task is done or marking a task or loading data!

senwei01 commented on src/main/java/duke/command/DateCommand.java:
```diff
@@ -0,0 +1,47 @@
+     * @param storage To save the list after making changes.
+     */
+    @Override
+    public void execCommand(TaskList list, Ui ui, Storage storage) {
```
I like the OOP of this!

senwei01 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,59 @@
+package duke;
```
I like how you have multiple ways to filter tasks :)

senwei01 reviewed (commented):
Overall, I like your code! It adheres to all the coding standards, and the OOP is really great!

Yingming23 committed:
Added GUI

Yingming23 committed:
Merge branch 'branch-Level-10'

WuHaohui1231 commented:
Your code is excellent overall. It follows the OOP principles. Some implementation of the increments are well-coded and smart.

WuHaohui1231 reviewed (commented):
The feature is well-implemented.

WuHaohui1231 commented on src/main/java/Input.java:
```diff
@@ -93,6 +116,8 @@ public Command getCommand() throws DukeException {
+            return new DateCommand(getDate());
         } else {
             throw new DukeException("Unknown command. Please enter a valid command");
         }
```
Can consider reformatting this part, as there are many indents.

Yingming23 committed:
Parser and TaskList class: added assert commands

No assert commands.

Adding assert commands help to check the inputs for illegal actions.

Let's add assert commands in the code to prevent errors.

Using assert will be more safe.

Yingming23 committed:
Improved code quality

Yingming23 committed:
Merge pull request #1 from Yingming23/branch-A-Assertions

Parser and TaskList class: added assert commands

Yingming23 committed:
Merge branch 'master' of https://github.com/Yingming23/ip

Yingming23 committed:
Merge branch 'master' into Branch-A-CodeQuality

Yingming23 committed:
Merge pull request #2 from Yingming23/Branch-A-CodeQuality

Yingming23 committed:
Add UpdateCommand: New command to update tasks

No command to update tasks

A command to update tasks is useful as it will allow users to
change task details or time without deleting and adding again

Let's add the UpdateCommand to allow users ease of use.

Yingming23 committed:
Merge branch 'branch-C-Update'

Yingming23 committed:
GUI updated to be more polished and with more personality

Yingming23 committed:
Added Ui.png

Yingming23 committed:
Update README.md

Edfernape commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,31 @@
+package duke;
```
very clean and nice main program!

Edfernape reviewed (commented):
your code is really clean
